src/internal/_xtake.js

```diff
@@ -15,7 +15,8 @@ module.exports = (function() {
       return _reduced(result);
     } else {
       this.n -= 1;
-      return this.xf['@@transducer/step'](result, input);
+      return this.n === 0 ? _reduced(this.xf['@@transducer/step'](result, input))
+                          : this.xf['@@transducer/step'](result, input);
     }
```

Comment:
Have both an `n == 0` case and an `n == 1` case is unfortunate. Is there any way to simplify this? There may not be.